paddle/operators/beam_search_op.h

```diff
@@ -205,7 +206,7 @@ class BeamSearchOp : public framework::OperatorBase {
     size_t beam_size = Attr<int>("beam_size");
     int end_id = Attr<int>("end_id");
     LOG(INFO) << "init beam search";
-    BeamSearch alg(ids, scores, level, beam_size, end_id);
+    BeamSearch<size_t, float> alg(ids, scores, level, beam_size, end_id);
```

Comment:
Here acctually assume the input data is size_t and float.

Please reference [here](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/increment_op.cc#L64) to fix it.